src/Sentry.Unity/Integrations/UnityApplicationLoggingIntegration.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using Sentry.Integrations;
+using UnityEngine;
+
+namespace Sentry.Unity.Integrations
+{
+    internal sealed class UnityApplicationLoggingIntegration : ISdkIntegration
+    {
+        internal readonly ErrorTimeDebounce ErrorTimeDebounce = new(TimeSpan.FromSeconds(1));
+        internal readonly LogTimeDebounce LogTimeDebounce = new(TimeSpan.FromSeconds(1));
+        internal readonly WarningTimeDebounce WarningTimeDebounce = new(TimeSpan.FromSeconds(1));
+
+        // TODO: remove 'IEventCapture' in  further iteration
+        private readonly IEventCapture? _eventCapture;
+        private readonly IAppDomain _appDomain;
+
+        private IHub? _hub;
+
+        public UnityApplicationLoggingIntegration(IAppDomain? appDomain = null, IEventCapture? eventCapture = null)
+        {
+            _appDomain = appDomain ?? UnityAppDomain.Instance;
+            _eventCapture = eventCapture;
+        }
+
+        public void Register(IHub hub, SentryOptions _)
+        {
+            _hub = hub;
+
+            _appDomain.LogMessageReceived += OnLogMessageReceived;
+            _appDomain.Quitting += OnQuitting;
+        }
+
+        // Internal for testability
+        internal void OnLogMessageReceived(string condition, string stackTrace, LogType type)
+        {
+            var debounced = type switch
+            {
+                LogType.Error or LogType.Exception or LogType.Assert => ErrorTimeDebounce.Debounced(),
+                LogType.Log => LogTimeDebounce.Debounced(),
+                LogType.Warning => WarningTimeDebounce.Debounced(),
+                _ => true
+            };
+            if (!debounced)

```

Comment:
Lets check `_hub` for null once here to avoid the `_hub?.` below this point:

```suggestion
            if (!debounced || _hub is null)

```